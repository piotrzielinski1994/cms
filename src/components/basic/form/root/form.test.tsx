import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import Form from './form';

describe('Form', () => {
  const component = (
    <Form.Root>
      <Form.Group>
        <Form.Label htmlFor="id">Label</Form.Label>
        <input id="id" />
        <Form.Error>Error</Form.Error>
      </Form.Group>
    </Form.Root>
  );

  it('should have no accessibility violations', async () => {
    const { container } = render(component);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
