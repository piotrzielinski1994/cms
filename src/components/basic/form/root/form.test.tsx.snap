// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > should match the snapshot 1`] = `
<div>
  <form>
    <div
      class="grid gap-1 content-start grid-rows-[auto_1fr]"
    >
      <label
        class="self-end"
        for="id"
      >
        Label
      </label>
      <input
        id="id"
      />
      <span
        aria-hidden="false"
        class="min-h-[1em] text-sm text-red-500 leading-none"
        role="alert"
      >
        Error
      </span>
    </div>
  </form>
</div>
`;
